ReactAndroid/src/main/java/com/facebook/react/uimanager/animation/NativeAnimatedNodesManager.java

```diff
@@ -0,0 +1,344 @@
+        "already attached to a view");
+    }
+    propsAnimatedNode.mConnectedViewTag = viewTag;
+  }
+
+  public void disconnectAnimatedNodeFromView(int animatedNodeTag, int viewTag) {
+    AnimatedNode node = mAnimatedNodes.get(animatedNodeTag);
+    if (node == null) {
+      throw new JSApplicationIllegalArgumentException("Animated node with tag " + animatedNodeTag +
+        " does not exists");
+    }
+    if (!(node instanceof PropsAnimatedNode)) {
+      throw new JSApplicationIllegalArgumentException("Animated node connected to view should be" +
+        "of type " + PropsAnimatedNode.class.getName());
+    }
+    PropsAnimatedNode propsAnimatedNode = (PropsAnimatedNode) node;
+    if (propsAnimatedNode.mConnectedViewTag == viewTag) {
+      propsAnimatedNode.mConnectedViewTag = -1;
+    }
+  }
+
+  /**
+   * Animation loop performs two DFSes over the graph of animated nodes. We use incremented
+   * {@code mAnimatedGraphDFSColor} to mark nodes as visited in each of the DFSes which saves
+   * additional loops for clearing "visited" states.
+   *
+   * First DFS starts with nodes that are in {@code mUpdatedNodes} or directly attached to an active
+   * animation (hence linked to objects from {@code mActiveAnimations}). In that step we calculate
+   * an attribute {@code mActiveIncomingNodes}. The second DFS runs in topological order over the
+   * sub-graph of *active* nodes. This is done by adding node to the DFS queue only if all its
+   * "predecessors" have already been visited.
+   */
+  private void runAnimationStep(long frameTimeNanos) {
+    int activeNodesCount = 0;
+    int updatedNodesCount = 0;
+    boolean hasFinishedAnimations = false;
+
+    // STEP 1.
+    // DFS over graph of nodes starting from ones from `mUpdatedNodes` and ones that are attached to
+    // active animations (from `mActiveANimations`. Update `mIncomingNodes` attribute for each node
+    // during that DFS. Store number of visited nodes in `activeNodesCount`. We "execute" active
+    // animations as a part of this step.
+
+    mAnimatedGraphDFSColor++; /* use new color */
+    if (mAnimatedGraphDFSColor == AnimatedNode.INITIAL_DFS_COLOR) {
+      // value "0" is used as an initial color for a new node, using it in DFS may cause some nodes
+      // to be skipped.
+      mAnimatedGraphDFSColor++;
+    }
+
+    Queue<AnimatedNode> nodesQueue = new ArrayDeque<>();
+    for (int i = 0; i < mUpdatedNodes.size(); i++) {
+      AnimatedNode node = mUpdatedNodes.get(i);
+      if (node.mDFSColor != mAnimatedGraphDFSColor) {
+        node.mDFSColor = mAnimatedGraphDFSColor;
+        activeNodesCount++;
+        nodesQueue.add(node);
+      }
+    }
+
+    for (int i = 0; i < mActiveAnimations.size(); i++) {
+      AnimationDriver animation = mActiveAnimations.get(i);
+      animation.runAnimationStep(frameTimeNanos);
+      AnimatedNode valueNode = animation.mAnimatedValue;
```

Comment:
IMO there is no use-case for such an action. Currently it will not be possible as `setValue` from JS will actually stop active animation (if any) (see [AnimatedImplementation.js:553](https://github.com/facebook/react-native/blob/11985d5c7721b0f9c9f46a04515799db0eeb44eb/Libraries/Animated/src/AnimatedImplementation.js#L553))